crates/bevy_ecs/src/world/mod.rs

```diff
@@ -0,0 +1,819 @@
+                    ArchetypeComponentInfo {
+                        archetype_component_id: ArchetypeComponentId::new(
+                            *archetype_component_count,
+                        ),
+                        storage_type: StorageType::Table,
+                    },
+                );
+                *archetype_component_count += 1;
+                let component_info = components.get_info_unchecked(component_id);
+                Column::with_capacity(component_info, 1)
+            })
+    }
+
+    pub(crate) fn initialize_resource<T: Component>(&mut self) -> ComponentId {
+        let component_id = self.components.get_or_insert_resource_id::<T>();
+        // SAFE: resource initialized above
+        unsafe { self.initialize_resource_internal(component_id) };
+        component_id
+    }
+
+    pub(crate) fn initialize_non_send_resource<T: 'static>(&mut self) -> ComponentId {
+        let component_id = self.components.get_or_insert_non_send_resource_id::<T>();
+        // SAFE: resource initialized above
+        unsafe { self.initialize_resource_internal(component_id) };
+        component_id
+    }
+
+    /// returns the resource column if the requested resource exists
+    pub(crate) fn get_populated_resource_column(
+        &self,
+        component_id: ComponentId,
+    ) -> Option<&Column> {
+        let resource_archetype = self.archetypes.resource();
+        let unique_components = resource_archetype.unique_components();
+        unique_components.get(component_id).and_then(|column| {
+            if column.is_empty() {
+                None
+            } else {
+                Some(column)
+            }
+        })
+    }
+
+    fn validate_non_send_access<T: 'static>(&self) {
+        if !self.main_thread_validator.is_main_thread() {
+            panic!(
+                "attempted to access NonSend resource {} off of the main thread",
+                std::any::type_name::<T>()
+            );
+        }
+    }
+
+    /// Empties queued entities and adds them to the empty [Archetype].
+    /// This should be called before doing operations that might operate on queued entities,
+    /// such as inserting a [Component].
+    pub(crate) fn flush(&mut self) {
+        let empty_archetype = self.archetypes.empty_mut();
+        unsafe {
+            // SAFE: archetype tables always exist
+            let table = self
+                .storages
+                .tables
+                .get_unchecked_mut(empty_archetype.table_id());
+            // PERF: consider pre-allocating space for flushed entities
+            self.entities.flush(|entity, location| {
+                // SAFE: no components are allocated by archetype.allocate() because the archetype is empty
+                *location = empty_archetype.allocate(entity, table.allocate(entity));
+            });
+        }
+    }
+}
+
+impl fmt::Debug for World {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "World")
```

Comment:
Maybe offer some simple statistics like entity count?

Reply:
Yes please! Number of archetypes and components / resources registered are the other things I would commonly want.